components/governance/org.wso2.carbon.governance.lcm/src/main/java/org/wso2/carbon/governance/lcm/tasks/LCNotificationScheduler.java

```diff
@@ -0,0 +1,323 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.governance.lcm.tasks;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.governance.api.exception.GovernanceException;
+import org.wso2.carbon.governance.lcm.tasks.events.LifecycleNotificationEvent;
+import org.wso2.carbon.governance.lcm.util.CommonUtil;
+import org.wso2.carbon.governance.lcm.util.LifecycleStateDurationUtils;
+import org.wso2.carbon.governance.registry.extensions.aspects.utils.LifecycleConstants;
+import org.wso2.carbon.registry.common.AttributeSearchService;
+import org.wso2.carbon.registry.common.ResourceData;
+import org.wso2.carbon.registry.common.eventing.NotificationService;
+import org.wso2.carbon.registry.core.ResourceImpl;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.eventing.services.EventingServiceImpl;
+import org.wso2.carbon.registry.indexing.IndexingConstants;
+
+import javax.xml.namespace.QName;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * This Util class holds methods to send checkpoint notifications and add schedulers.
+ */
+public class LCNotificationScheduler {
+
+    private static final Log log = LogFactory.getLog(LCNotificationScheduler.class);
+
+    // Duration of a day 1 * 24 * 60 * 60 * 1000.
+    private final long durationOfADay = 86400000;
+
+    // Used to as Qname for checkpoint id.
+    private final String checkpointId = "id";
+
+    // Date separator.
+    private final String dateSeparator = "-";
+
+    // Date length.
+    private final int dateLength = 10;
+
+    // Date format used in MySQL queries.
+    private final String dateFormat = "yyyy-M-d";
+
+    // Lifecycle property name to which is used to store.
+    private final String lcCheckPointPropertyName = "registry.LCCheckpoints";
+
+    // Super admin username.
+    private final String superAdminUsername = "admin";
+
+    // Super tenant domain.
+    private final String superTenantDomain = "carbon.super";
+
+    // Super tenant tenant Id.
+    private final int superTenantId = -1234;
+
+    // Equals constant used for solar advance search.
+    private final String equals = "eq";
+
+    // Not applicable constant used for solar advance search.
+    private final String notApplicable = "na";
+
+    /**
+     * This method is used to send notifications for checkpoints in lifecycle. This method is called using the
+     * scheduler task CheckpointNotificationSchedulerTask.
+     */
+    public void run() {
```

Comment:
HI,

Made run method protected in pull request [1], commit[2].
[1] - https://github.com/wso2/carbon-governance/pull/44
[2] - https://github.com/prasa7/carbon-governance/commit/c681dda86c2df780a044e475a55cd5a4b2f5ef89

Thanks